src/jit/codegencommon.cpp

```diff
@@ -10284,12 +10284,21 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     ScopedSetVariable<bool> _setGeneratingProlog(&compiler->compGeneratingProlog, true);
 
+    gcInfo.gcResetForBB();
+
     compiler->unwindBegProlog();
 
-    // TODO Save callee-saved registers
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
+
+    inst_RV(INS_push, REG_EBX, TYP_REF);
+    inst_RV(INS_push, REG_ESI, TYP_REF);
+    inst_RV(INS_push, REG_EDI, TYP_REF);
 
```

Comment:
It would be better to use ``genPushCalleeSavedRegisters`` instead of explicitly adding push instructions.